### ISSUE TITLE
Configure workflows to run for webhook events (i.e. check_run, check_suite, deployment, etc.)

### ISSUE THREAD

**Issue by lfarci** [lfarci/github-actions#5]

**Comment by lfarci** [comment]
I had some trouble to understand how `check_run` and `check_suite` and `deployment` were related to webhooks. Now I think I get it. In short, external systems or GitHub actions developed by third-parties are able to create resources using the GitHub API (create or interact by the way). When marking workflows with this kind of event, we are actually looking to run them in response to external events (webhooks).

**Check runs**
Workflows triggered by those event runs in response to a webhook that might be triggered by a third party GitHub action or another system. The only condition being to create a new check run using the GitHub API.

<details><summary>How are check runs events related to webhooks?</summary>
<p>
`check_run` and `check_suite` are two events that can trigger workflows. They are checks performed on the code. It can be linters, tests, static analysis or anything. They are commonly run in pull requests but they can be executed at any steps during the build process. 

While it's possible to create and manage check runs directly from a GitHub Actions workflow using the GitHub API, it's more common to use existing actions that abstract away these details.

For example, many actions will automatically create check runs for their operation. This is the case for actions that run tests, linters, or other checks on your code. These actions will create a check run, execute the check, and then update the check run with the result. This allows you to see the result of the check directly in the GitHub UI, without having to manually create and manage the check run.

In other cases, check runs might be created by external services that integrate with GitHub. For example, a continuous integration (CI) service might create a check run for each build it performs. This allows you to see the status of the build directly in GitHub, even though the build is performed by an external service.

So while it's possible to manually create and manage check runs from a GitHub Actions workflow, it's more common to use actions or services that handle these details for you.
</p>
</details> 

**Deployments**
This is similar to check runs events. Any external system or GitHub actions can create a new deployment in the workflow's directory. The `deployment` event captures this and starts a workflow.

**`repository_dispatch`**
The `repository_dispatch` event is a way to trigger a workflow directly, without needing to create or modify resources like check runs or deployments.

The `repository_dispatch` event is not triggered automatically by GitHub. Instead, it's designed to be triggered manually from an external source via a webhook. This makes it a flexible way to integrate GitHub Actions with external systems.

To trigger a `repository_dispatch` event, you send a POST request to the repository's dispatches endpoint. This request must be authenticated using a personal access token with the `repo` scope.

Here's an example of a workflow that runs on a `repository_dispatch` event:

```yaml
name: My Workflow
on:
  repository_dispatch:
    types: [my-event]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      - name: Run a script
        run: echo "Hello, world!"
```

In this example, the workflow will run whenever a `repository_dispatch` event with the type `my-event` is created.

So in summary, while many workflow triggers in GitHub Actions are related to GitHub resources and events, the `repository_dispatch` event provides a way to trigger workflows directly from an external system, without needing to create or modify GitHub resources.
